pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -1016,14 +1050,7 @@ func (o *MustGatherOptions) newPod(node, image string, hasMaster bool) *corev1.P
 			HostNetwork:                   o.HostNetwork,
 			NodeSelector:                  nodeSelector,
 			TerminationGracePeriodSeconds: &zero,
-			Tolerations: []corev1.Toleration{
-				{
-					// An empty key with operator Exists matches all keys,
-					// values and effects which means this will tolerate everything.
-					// As noted in https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
-					Operator: "Exists",
-				},
-			},
+			Tolerations:                   filteredTolerations,
```

Comment:
There are two case to consider. Once for when `node == "" && hasMaster` is true. And the other case. So either the list of tolerations has both tolerations or just the one with `"node.kubernetes.io/not-ready"`.